resolver/resolver.cc

```diff
@@ -1185,7 +1185,7 @@ class ResolveConstantsWalk {
                         }
 
                         if constexpr (isMutableStateType) {
```

Comment:
You no longer need this `if` guard, because neither `.data(gs)` nor `lookupSingletonClass(gs)` require that `gs` be mutable.